File: src/models/Candidate.js
export class Candidate {
  constructor(data = {}) {
    this.id = data.id || null;
    this.name = data.name || '';
    this.email = data.email || '';
    this.phone = data.phone || '';
    this.college = data.college || '';
    this.branch = data.branch || '';
    this.gender = data.gender || '';
    this.backlogs = data.backlogs || 0;
    // resumeData is stored separately, resumeName is stored in JSON
    this.resumeName = data.resumeName || null;
  }
}

File: src/main/java/com/jobportel/boot/model/Candidate.java
package com.jobportel.boot.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Data;

@Data
public class Candidate {
    private Long id;
    private String name;
    private String email;
    private String phone;
    private String college;
    private String branch;
    private String gender;
    private int backlogs;
    @JsonIgnore
    private byte[] resumeData;
    private String resumeName;
}